source/javascripts/components/_AddStep.js.erb

```diff
@@ -463,14 +461,32 @@ angular.module("BitriseWorkflowEditor").service("addStepService", function() {
 		}
 	};
 
-}).directive("addStepSidebarStep", function($animate) {
+}).directive("addStepSidebarStep", function($filter, addStepService, imageService) {
 
 	return {
 		restrict: "C",
 		templateUrl: "<%= mode_dependant_asset_path(data[:routes][:templates][:add_step_sidebar_step] )%>",
 		scope: {
 			step: "=",
 			selected: "&selectCallback"
+		},
+		link: function(scope, element) {
+			scope.addStepService = addStepService;
+			addStepService.stepIntersectionObserverCallbacks[scope.step.id] = function() {
```

Comment:
should not we hide `stepIntersectionObserver` in the `addStepService`? We could expose an `observer` and `unobserve` method from the service, but how that does it only concerns the service not this component. So I'd recommend implement something like this: `addStepService.objserve(element[0], function callback() { .... })` and `stepService.unobserve(element[0])`;